sencha-workspace/packages/slate-cbl/src/view/teacher/DashboardController.js

```diff
@@ -222,11 +242,145 @@ Ext.define('Slate.cbl.view.teacher.DashboardController', {
             });
         });
     },
+    
+    onDemonstrationDelete: function(demonstration, operation) {
+        var me = this,
+            dashboardView = me.view,
+            mainGridEl = dashboardView.el.down('.cbl-grid-main'),
```

Comment:
this whole method is a sloppy copy of onDemonstrationSave intended to get the page updating when demonstrations are deleted. It should be de-duplicated and maybe an abstract method or two broken out